src/coreclr/jit/importer_vectorization.cpp

```diff
@@ -6,16 +6,43 @@
 #pragma hdrstop
 #endif
 
+//------------------------------------------------------------------------
+// importer_vectorization.cpp
+//
+// This file is responsible for various (partial) vectorizations during import phase,
+// e.g. the following APIs are currently supported:
+//
+//   1) String.Equals(string, string)
+//   2) String.Equals(string, string, StringComparison.Ordinal)
+//   3) str.Equals(string)
+//   4) str.Equals(String, StringComparison.Ordinal)
+//   5) str.StartsWith(string, StringComparison.Ordinal)
+//   6) MemoryExtensions.SequenceEqual<char>(ROS<char>, ROS<char>)
+//   7) MemoryExtensions.Equals(ROS<char>, ROS<char>, StringComparison.Ordinal)
+//   8) MemoryExtensions.StartsWith<char>(ROS<char>, ROS<char>)
+//   9) MemoryExtensions.StartsWith(ROS<char>, ROS<char>, StringComparison.Ordinal)
+//
+// When one of the arguments is a constant string of a [0..32] size so we can inline
+// a vectorized comparison against it using SWAR or SIMD techniques (e.g. via two V256 vectors)
+//
+// We might add these in future:
+//   1) OrdinalIgnoreCase for everything above
+//   2) Span.CopyTo
+//   3) Spans/Arrays of bytes (e.g. UTF8) against a constant RVA data
+//
+
 //------------------------------------------------------------------------
 // impExpandHalfConstEqualsSIMD: Attempts to unroll and vectorize
 //    Equals against a constant WCHAR data for Length in [8..32] range
 //    using SIMD instructions. It uses the following expression:
 //
-//      bool equasl = ((v1 ^ cns1) | (v2 ^ cns2)) == Vector128.Zero
+//       bool equasl = ((v1 ^ cns1) | (v2 ^ cns2)) == Vector128.Zero
 //
 //    or if a single vector is enough (len == 8 or len == 16 with AVX):
 //
-//      bool equasl = (v1 ^ cns1) == Vector128.Zero
+//       bool equasl = (v1 ^ cns1) == Vector128.Zero
+//
+//    C# equivalent of what this function emits: https://gist.github.com/EgorBo/085f9c15da11eecc00f482710f48aef6
```

Comment:
Inline this code snippet instead of linking to your private gist that has a good chance of disappearing in a few years?